v2/test/presubmit-v2-go-test.sh

```diff
@@ -35,7 +34,17 @@ cd /home/prow/go/src/github.com/kubeflow/pipelines/v2
 ${GO_CMD} mod download
 ${GO_CMD} mod tidy
 git diff --exit-code -- go.mod go.sum || (echo "go modules are not tidy, run 'go mod tidy'." && exit 1)
+
 # Note, for tests that use metadata grpc api, port-forward it locally in a separate terminal by:
-kubectl port-forward svc/metadata-grpc-service 8080:8080 -n $NAMESPACE
-# 3. run test in project directory
+if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]]; then
+  gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+fi
+gcloud container clusters get-credentials "$TEST_CLUSTER" --region "$REGION" --project "$PROJECT"
+set +e
+kubectl port-forward svc/metadata-grpc-service 8080:8080 -n "$NAMESPACE" & PORT_FORWARD_PID=$!
+# wait for kubectl port forward
+sleep 10
 ${GO_CMD} test -v -cover ./...
+# kill kubectl port forward before exit
+echo "killing kubectl port forward before exit"
+kill "$PORT_FORWARD_PID"
```

Comment:
Will the Pod get stuck in pending if we don't kill the port forward?

If necessary, then this should be in a trap (a syntax in bash like exit handler)

Reply:
Done. (Previously I used `set +e` to make sure the kill command will get called. trap looks clearer.